camunda.Dockerfile

```diff
@@ -0,0 +1,166 @@
+
+# https://github.com/docker-library/openjdk/issues/212#issuecomment-420979840
+# https://openjdk.java.net/jeps/341
+# TL;DR generate some class data sharing for faster load time
+RUN java -Xshare:dump;
+
+### Build camunda from scratch ###
+FROM java as build
+WORKDIR /camunda
+ENV MAVEN_OPTS -XX:MaxRAMPercentage=80
+COPY --link . ./
+RUN --mount=type=cache,target=/root/.m2,rw \
+    ./mvnw -B -am -pl dist package -T1C -D skipChecks -D skipTests && \
+    mv dist/target/camunda-zeebe .
+
+### Extract camunda from distball ###
+# hadolint ignore=DL3006
+FROM base as distball
+WORKDIR /camunda
+ARG DISTBALL="dist/target/camunda-zeebe-*.tar.gz"
+COPY --link ${DISTBALL} camunda.tar.gz
+
+# Remove zbctl from the distribution to reduce CVE related maintenance effort w.r.t to containers
+RUN mkdir camunda-zeebe && \
+    tar xfvz camunda.tar.gz --strip 1 -C camunda-zeebe && \
+    find . -type f -name 'zbctl*' -delete
+
+### Image containing the camunda distribution ###
+# hadolint ignore=DL3006
+FROM ${DIST} as dist
+
+### Application Image ###
+# https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+# hadolint ignore=DL3006
+FROM java as app
+# leave unset to use the default value at the top of the file
+ARG BASE_IMAGE
+ARG BASE_DIGEST
+ARG VERSION=""
+ARG DATE=""
+ARG REVISION=""
+
+# OCI labels: https://github.com/opencontainers/image-spec/blob/main/annotations.md
+LABEL org.opencontainers.image.base.digest="${BASE_DIGEST}"
+LABEL org.opencontainers.image.base.name="docker.io/library/${BASE_IMAGE}"
+LABEL org.opencontainers.image.created="${DATE}"
+LABEL org.opencontainers.image.authors="zeebe@camunda.com"
+LABEL org.opencontainers.image.url="https://zeebe.io"
+LABEL org.opencontainers.image.documentation="https://docs.camunda.io/docs/self-managed/zeebe-deployment/"
+LABEL org.opencontainers.image.source="https://github.com/camunda/camunda"
+LABEL org.opencontainers.image.version="${VERSION}"
+# According to https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys
+# and given we set the base.name and base.digest, we reference the manifest of the base image here
+LABEL org.opencontainers.image.ref.name="${BASE_IMAGE}"
+LABEL org.opencontainers.image.revision="${REVISION}"
+LABEL org.opencontainers.image.vendor="Camunda Services GmbH"
+LABEL org.opencontainers.image.licenses="(Apache-2.0 AND LicenseRef-Zeebe-Community-1.1)"
+LABEL org.opencontainers.image.title="Camunda 8"
+LABEL org.opencontainers.image.description="Workflow engine for microservice orchestration"
+
+# OpenShift labels: https://docs.openshift.com/container-platform/4.10/openshift_images/create-images.html#defining-image-metadata
+LABEL io.openshift.tags="bpmn,orchestration,workflow"
+LABEL io.k8s.description="Workflow engine for microservice orchestration"
+LABEL io.openshift.non-scalable="false"
+LABEL io.openshift.min-memory="512Mi"
+LABEL io.openshift.min-cpu="1"
```

Comment:
I need your help to put the right labels

Reply:
Those are only minimum values so not sure how often they are used in practice. I assume memory usage goes up with the single application to bumping that to `1024Mi` (or even more?) would be good.

Reply:
I need to put valid values to the other labels: `org.opencontainers.*` , `io.openshift.*` , `io.k8s.description`